gobblin-data-management/src/main/java/org/apache/gobblin/data/management/retention/dataset/CleanableDatasetStoreDataset.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gobblin.data.management.retention.dataset;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Properties;
+import org.apache.gobblin.data.management.version.TimestampedDatasetStateStoreVersion;
+import org.apache.gobblin.data.management.version.finder.TimestampedDatasetStateStoreVersionFinder;
+import org.apache.gobblin.data.management.version.finder.VersionFinder;
+import org.apache.gobblin.metastore.DatasetStoreDataset;
+import org.apache.hadoop.fs.FileSystem;
+import lombok.Data;
+
+
+/**
+ * A cleanable {@link DatasetStoreDataset}
+ */
+@Data
+public class CleanableDatasetStoreDataset extends ModificationTimeDataset {
```

Comment:
We probably don't want to assume the CleanableDatasetStoreDataset is always using Timebased version finder.